### PR TITLE
http_client.c: Fix inconsistency w.r.t. type of max_resp_len

### PR DESCRIPTION
**Pull request by DDvO**
See https://github.com/openssl/openssl/commit/8f965908a53b4f0c5a735739e8a273a3a33a976e#commitcomment-50727959.

### CODE REVIEW

File: crypto/http/http_client.c [hunk 1]
```diff
@@ -135,7 +135,7 @@ void OSSL_HTTP_REQ_CTX_set_max_response_length(OSSL_HTTP_REQ_CTX *rctx,
         ERR_raise(ERR_LIB_HTTP, ERR_R_PASSED_NULL_PARAMETER);
         return;
     }
-    rctx->max_resp_len = len != 0 ? len : HTTP_DEFAULT_MAX_RESP_LEN;
+    rctx->max_resp_len = len != 0 ? (size_t)len : HTTP_DEFAULT_MAX_RESP_LEN;
```
Comment (t8m) [inline crypto/http/http_client.c:138]: I suppose the signature of SSL_HTTP_REQ_CTX_set_max_response_length should be changed to size_t as well?

Comment (DDvO) [reply crypto/http/http_client.c:138]: I suppose we need to keep the type of `OSSL_HTTP_REQ_CTX_set_max_response_length()` for backward compatibility, so deliberately did not update that.

Comment (t8m) [reply crypto/http/http_client.c:138]: Ah, right, it is used by the old OCSP call macro. That could be made a deprecated wrapper function instead.

Comment (t8m) [reply crypto/http/http_client.c:138]: Not sure it is worth it though.

Comment (DDvO) [reply crypto/http/http_client.c:138]: I don't this it's worth the effort.
People are anyway better off using the high-level HTTP API,
and the difference between the old `unsigned long` and the nicer `size_t` should be negligible in practice.

**Review by t8m — APPROVED** [review]

**Review by levitte — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
Wow, this innocent PR meanwhile has 3 approvals, 
one by an OTC members and two by OMS members - my personal record :smile:
Can I convert the extra approvals to shopping bonus points, frequent-flyer miles, or bitcoins? :rofl:

**Comment by paulidale** [comment]
There is a PR in one of the repositories that was approved by the entire OMC.

**Comment by DDvO** [comment]
I just tried it out: `ghmerge`/`addrev` can take any number of reviewers and eliminates duplicates.

**Comment by DDvO** [comment]
What actually disappoints me regards the finding handled by this PR:
Despite all the pedantic settings of the various compilers used by the CI tests, 
apparently none of them flagged the potential discrepancy between `unsigned long` and `size_t`.
Is there a way to add such a check to the CI?
Or would the Microsoft `cl` detect this, which is currently (mostly) disabled?

**Comment by t8m** [comment]
AppVeyor builds (still running on pushes to master) currently fail due to this.

**Comment by DDvO** [comment]
Good to know that AppVeyor would have caught this inconsistency.

**Comment by paulidale** [comment]
I think this was meant to be closed....

**Pull request closed without merging**